lib/identity_cache/configuration_dsl.rb

```diff
@@ -279,7 +279,7 @@ def attribute_dynamic_fetcher(attribute, fields, values) #:nodoc:
       end
 
       def add_parent_expiry_hook(options)
-        child_class = options[:association_class]
+        child_class = options[:association_class].klass
```

Comment:
If autoloading is disabled, then that to me indicates that the developer using the library would prefer to manually require the dependant model.